tests/unit/test_generate_build_matrix.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+from src.build_rock.configure.generate_build_matrix import (
+    get_target_archs,
+    configure_matrices,
+    MissingArchSupport,
+    set_build_config_outputs,
+)
+from pathlib import Path
+import pytest
+from tempfile import TemporaryDirectory
+
+
+def test_get_target_archs():
+    """Test extraction of target architectures from rockcraft project configuration"""
+    project = {
+        "name": "hello-world",
+        "title": "Hello World",
+        "summary": "An Hello World rock",
+        "description": "This is just an example of a Rockcraft project\nfor a Hello World rock.\n",
+        "version": "latest",
+        "base": "bare",
+        "build-base": "ubuntu@22.04",
+        "license": "Apache-2.0",
+        "run-user": "_daemon_",
+        "environment": {"FOO": "bar"},
+        "services": {
+            "hello": {
+                "override": "replace",
+                "command": "/usr/bin/hello -t",
+                "environment": {"VAR1": "value", "VAR2": "other value"},
+            }
+        },
+        "platforms": {
+            "amd64": None,
+            "armhf": {"build-for": ["armhf", "arm64"]},
+            "ibm": {"build-on": ["s390x"], "build-for": "s390x"},
+        },
+        "parts": {"hello": {"plugin": "nil", "stage-packages": ["hello"]}},
+    }
```

Comment:
We can use a fixture to store this constant string ;D

Reply:
Done. We now have a fixture for this at tests/fixtures/sample_data.py